backend/danswer/server/query_and_chat/token_budget.py

```diff
@@ -0,0 +1,45 @@
+import json
+from datetime import datetime, timedelta
+
+from fastapi import HTTPException, Depends
+from sqlalchemy.orm import Session
+from sqlalchemy import func
+
+from danswer.dynamic_configs import get_dynamic_config_store
+from danswer.db.models import ChatMessage
+from danswer.db.engine import get_session
+from danswer.configs.constants import ENABLE_TOKEN_BUDGET, TOKEN_BUDGET, TOKEN_BUDGET_TIME_PERIOD, TOKEN_BUDGET_SETTINGS
+
+BUDGET_LIMIT_DEFAULT = -1 # Default to no limit
+TIME_PERIOD_HOURS_DEFAULT = 12
+
+def is_under_token_budget(db_session: Session) -> bool:
+    settings_json = get_dynamic_config_store().load(TOKEN_BUDGET_SETTINGS)
+    settings = json.loads(settings_json)
+
+    is_enabled = settings.get(ENABLE_TOKEN_BUDGET, False)
```

Comment:
nit: should we check this first to avoid an extra call to `get_dynamic_config_store().load` if it's disabled anyways

Reply:
I'm not clear on how to fix this. `is_enabled` depends on `get_dynamic_config_store()`, so I would have to call it to get this value in the first place, right? What would the alternative look like?